helper/resource/teststep_providers_test.go

```diff
@@ -1871,6 +1874,70 @@ func TestTest_TestStep_ProviderFactories_Refresh_Inline(t *testing.T) {
 	})
 }
 
+func TestTest_TestStep_ProviderFactories_CopyWorkingDir_EachTestStep(t *testing.T) {
+	t.Parallel()
+
+	err := os.Setenv(plugintest.EnvTfAccPersistWorkingDir, "1")
+	if err != nil {
+		t.Fatalf("cannot set %s env var: %s", plugintest.EnvTfAccPersistWorkingDir, err)
+	}
```

Comment:
Similar to `t.TempDir()`, there's [`(*testing.T).Setenv()`](https://pkg.go.dev/testing#T.Setenv). Setting environment variables in tests should implicitly _not_ be parallelized since any other places that get/set the environment variable would result in race conditions.